lesson_26/api/java/api_app/src/main/java/com/codedifferently/lesson26/web/MediaItemsController.java

```diff
@@ -29,4 +39,35 @@ public GetMediaItemsResponse getItems() {
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
     return response;
   }
+
+  @PostMapping("/items")
+  public ResponseEntity<CreateMediaItemResponse> postItems(
+      @Valid @RequestBody CreateMediaItemRequest request) {
+    MediaItemRequest itemRequest = request.getItem();
+    MediaItem item = MediaItemRequest.asMediaItem(itemRequest);
+    library.addMediaItem(item, librarian);
+    MediaItemResponse response = MediaItemResponse.from(item);
+    return ResponseEntity.ok(CreateMediaItemResponse.builder().item(response).build());
+  }
+
+  @GetMapping("/items/{id}")
+  public ResponseEntity<GetMediaItemsResponse> getItemsId(@PathVariable String id) {
+    Set<MediaItem> items = library.search(SearchCriteria.builder().id(id).build());
+    if (items.isEmpty()) {
+      return ResponseEntity.notFound().build();
+    }
+    MediaItemResponse responseItem = items.stream().map(MediaItemResponse::from).iterator().next();
+    return ResponseEntity.ok(GetMediaItemsResponse.builder().item(responseItem).build());
+  }
+
+  @DeleteMapping("/items/{id}")
+  public ResponseEntity<Void> deletebById(@PathVariable UUID id) {
+    Set<MediaItem> items = library.search(SearchCriteria.builder().id(id.toString()).build());
+    MediaItem item = items.iterator().next();
```

Comment:
This will explode if `items.isEmpty()` is true (meaning the collection is empty), so this should go after the if statement.